src/clipping/sutherlandhodgman.jl

```diff
@@ -18,10 +18,17 @@ The Sutherland-Hodgman algorithm for clipping polygons.
 """
 struct SutherlandHodgmanClipping <: ClippingMethod end
 
-function clip(poly::Polygon, other::Geometry, method::SutherlandHodgmanClipping)
-  c = [clip(ring, boundary(other), method) for ring in rings(poly)]
-  r = [r for r in c if !isnothing(r)]
-  isempty(r) ? nothing : PolyArea(r)
+# function clip(poly::Polygon, other::Geometry, method::SutherlandHodgmanClipping)
+  # c = [clip(ring, boundary(other), method) for ring in rings(poly)]
+  # r = [r for r in c if !isnothing(r)]
+  # isempty(r) ? nothing : PolyArea(r)
+# end
+
+
+function clip(poly::Polygon, other::Ring, method::SutherlandHodgmanClipping)
+ c = [clip(ring, other, method) for ring in rings(poly)]
+ r = [r for r in c if !isnothing(r)]
+ isempty(r) ? nothing : PolyArea(r)
 end
```

Comment:
Based on the latest comments above, I believe that the safest strategy in this PR is to confine the changes to the new clipping method, with minimum changes to other methods, and no addition of higher-level methods.

Can we rework the PR so that it only adds methods with types `Ring` and `Polygon`?  If a method with `Multi` is needed to handle some specific feature of `WeilerAthertonClipping`, that is fine too.

That way we can work on a separate PR later to better organize the dispatch with different clipping methods and geometries. Appreciate your patience improving the situation here :)